generic-templates/heartbeat-survey.html

```diff
@@ -0,0 +1,104 @@
+{#
+Snippet for prompting users with a UITour survey
+
+Variables:
+
+  @message - small text - Survey question text
+  @thankyouMessage - small text - Thanks message text shown after the user
+    interacts with the prompt
+  @engagementURL - URL - URL to open in a new tab after the user interacts with
+    the prompt
+  @engagementButtonLabel - small text - If blank, show a 5-star rating in the
+    prompt, otherwise show a button with this text
+  @learnMoreURL - URL - URL to open when the Learn More link is clicked
+  @learnMoreLabel - small text - Label for the Learn More link
+
+#}
+<div class="snippet" id="surveysnippet-{{ snippet_id }}"></div>
+<script type="text/javascript">
+//<![CDATA[
+(function() {
+  var ID = 'surveysnippet-{{ snippet_id }}';
+  var snippet = document.getElementById(ID);
+
+  snippet.addEventListener('show_snippet', function() {
+    if (gSnippetsMap.get(ID) !== 'shown' || gSnippetsMap.get('surveysnippet-override')) {
+      gSnippetsMap.set(ID, 'shown');
+
+      annotateUrl('{{ engagementURL|safe }}').then(function(engagementURL) {
+        document.dispatchEvent(new CustomEvent('mozUITour', {
+          bubbles: true,
+          detail: {
+    	      action: 'showHeartbeat',
+    	      data: {
+              message: '{{ message|safe }}',
```

Comment:
Indeed you need to escape this as @jpetto pointed:
`{{ message|escapejs|safe }}`